### PR TITLE
Enable dumping profiling information to timeline format viewable by chrome tracing.

### PR DESCRIPTION
**Pull request by alanarenda**

### CODE REVIEW

File: test/runtest.py [hunk 1]
```diff
@@ -1629,4 +1643,4 @@ def method(self):
 
 
 if __name__ == "__main__":
-  unittest.main(verbosity=2)
\ No newline at end of file
+  unittest.main(verbosity=2)
```
Comment (robertnishihara) [inline test/runtest.py]: Linting requires a new line at the end of each file.

File: test/runtest.py [hunk 2]
```diff
@@ -1598,6 +1598,20 @@ def f():
 
     ray.worker.cleanup()
 
+  def testWorkers(self):
```
Comment (robertnishihara) [inline test/runtest.py]: Each API method should be tested.

File: python/ray/experimental/state.py [hunk 3]
```diff
@@ -402,62 +411,75 @@ def micros(ts):
...
+      }
       full_trace.append(task)
 
-    with open(path, 'w') as outfile:
+    with open(path, "w") as outfile:
       json.dump(full_trace, outfile)
     task_info
 
-  def get_times(self, data):
+  def _get_times(self, data):
```
Comment (robertnishihara) [inline python/ray/experimental/state.py]: Methods that are not part of the API should be hidden from users (via a leading underscore). Note that this prevents the method from being shown by IPython's tab completion.

File: python/ray/experimental/state.py [hunk 4]
```diff
@@ -402,62 +411,75 @@ def micros(ts):
...
+        parent_trace = {
+            "cat": "submit_task",
+            "pid": "Node " + str(parent_worker["node_ip_address"]),
+            "tid": parent_info["worker_id"],
+            "ts": micros(min(parent_times)),
+            "ph": "s",
+            "name": "SubmitTask",
+            "args": {},
+            "id": str(worker)
+        }
```
Comment (robertnishihara) [inline python/ray/experimental/state.py]: Defining the dictionaries this way felt cleaner to me.

File: python/ray/experimental/state.py [hunk 5]
```diff
@@ -388,12 +388,21 @@ def task_profiles(self):
...
   def dump_catapult_trace(self, path):
+    """Dump task profiling information to a file.
+
+    This information can be viewed as a timeline of profiling information by
+    going to chrome://tracing in the chrome web browser and loading the
+    appropriate file.
+
+    Args:
+      path: The filepath to dump the profiling information to.
+    """
```
Comment (robertnishihara) [inline python/ray/experimental/state.py]: In general we should document python functions.

File: python/ray/rllib/policy_gradient/example.py [hunk 6]
```diff
@@ -35,6 +35,6 @@
   while result.training_iteration < config["max_iterations"]:
     print("\n== iteration", result.training_iteration)
     ray.global_state.dump_catapult_trace(
-      "/tmp/policy-gradient-%d.json" % result.training_iteration)
+        "/tmp/policy-gradient-%d.json" % result.training_iteration)
     result = alg.train()
-    print("current status: {}".format(result))
\ No newline at end of file
+    print("current status: {}".format(result))
```
Comment (robertnishihara) [inline python/ray/rllib/policy_gradient/example.py]: Linting requires a new line at the end of each file.

File: python/ray/rllib/policy_gradient/example.py [hunk 7]
```diff
@@ -35,6 +35,6 @@
   while result.training_iteration < config["max_iterations"]:
     print("\n== iteration", result.training_iteration)
     ray.global_state.dump_catapult_trace(
-      "/tmp/policy-gradient-%d.json" % result.training_iteration)
+        "/tmp/policy-gradient-%d.json" % result.training_iteration)
```
Comment (robertnishihara) [inline python/ray/rllib/policy_gradient/example.py]: This indentation failed the linter.

File: python/ray/experimental/state.py (hunk 5 above)
Comment (pcmoritz) [inline python/ray/experimental/state.py]: consider renaming it `dump_chrome_trace` (see `generate_chrome_trace_format` in TensorFlow); probably more people know chrome://tracing than know catapult

**Comment by robertnishihara** [comment]
This makes progress on #703.

**Pull request merged**